src/Native/System.Native/pal_readdir.h

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#include <dirent.h>
```

Comment:
DIR is defined as "typedef struct { ... } DIR;" which I don't think can be forward declared :(